dlib/vectorstream/vectorstream.h

```diff
@@ -43,6 +43,32 @@ namespace dlib
             {
                 read_pos = pos;
             }
+            
+            pos_type seekpos(pos_type pos, std::ios_base::openmode mode = std::ios_base::in | std::ios_base::out)
+            {
+                return seekoff(pos - pos_type(off_type(0)), std::ios_base::beg, mode);
```

Comment:
Are you getting a warning when you don't do `- pos_type(off_type(0))`?  I get what it's doing with regard to type conversion, but it really shouldn't be doing anything at runtime and seems unneeded.  But if it suppresses compiler warnings then this is fine.

Reply:
To be perfectly honest, a lot of stream objects implement ```seekpos``` like this, verbatim. So i copied it.